src/algorithms/newton_raphson.h

```diff
@@ -14,29 +14,54 @@ mat Family::newtonRaphson(const T& x, const mat& y, const double rho, const mat&
 
   Rcout.precision(4);
   uword p = x.n_cols;
+  uword n = x.n_rows;
   uword m = y.n_cols;
 
   mat z(u);
-  mat g(u);
+  mat g(size(z));
   mat h(p*m, p*m);
+  mat step(size(z));
 
   uword max_iter = 50;
-  double tolerance = 1e-12;
+  double tolerance = 1e-15;
   double alpha = 0.1;
   double gamma = 0.5;
 
+  // bool use_woodbury 
+  //   = (name() != "multinomial" && n < p)? true: false;
+
+  bool use_woodbury = false;
+
+  mat xxT;
+  vec activation;
+  
+  if (use_woodbury) {
+    xxT = x*x.t();
+  }
+
   uword iter = 0;
 
   while (iter < max_iter){
     
     mat lin_pred = x*z;
     
     g = gradient(x, y, lin_pred) + rho*(z-u);
-    h = hessian(x, y, lin_pred);
-
-    h.diag() += rho;
     
-    mat step = -reshape(solve(h,vectorise(g)),size(z));
+    if (use_woodbury) {
+      // activation = pseudoHessian(y, lin_pred);
+
+      // mat tmp = xxT;
+      // tmp.diag() += rho/activation;
+      // step = x.t()*solve(tmp, x*g) - g;
+      // step /= rho;
+
```

Comment:
should this really be commented out?

Reply:
I wanted to ask the same, it produces error due to the badly conditioned matrix. It won't get executed since use_woodbury is false, should I uncomment it?